src/filters.ts

```diff
@@ -1,46 +1,111 @@
+import { ScalarDefaults, FilterMappings } from './types'
+
+// If a filter is a scalar value, we find a default operator to map it to
+const scalarDefaults: ScalarDefaults = {
+  // this breaks BC as the original default as equalTo or like
+  string: 'like',
+  uuid: 'equalTo',
+  uuidarray: 'in',
+  bigint: 'equalTo',
+  bigintarray: 'in',
+  int: 'equalTo',
+  intarray: 'in'
+}
+
+const filterMappings: FilterMappings = {
+  bigint: {
+    equalTo: BigInt
+  },
+  bigintarray: {
+    in: (value: any[]) => value.map(BigInt)
+  },
+  int: {
+    equalTo: Number
+  },
+  intarray: {
+    in: (value: any[]) => value.map(Number)
+  },
+  intlistarray: {
+    contains: (value: any[]) => value.map(Number)
+  },
+  string: {
+    equalTo: String,
+    like: (value: any) => `%${value}%`,
+    likeInsensitive: (value: any) => `%${value}%`
```

Comment:
`like`/`likeInsensitive` should not default to using the value with the percentage symbols, because `includes` and `includesInsensitive ` already do that job.

I propose to turn this into:
```
    like: (value: any) => `${value}`,
    likeInsensitive: (value: any) => `${value}`,
    includes: (value: any) => `${value}`,
    includesInsensitive: (value: any) => `${value}`,
    ....
```

Which in turn should grow into the entire list of string operators defined here: https://github.com/graphile-contrib/postgraphile-plugin-connection-filter#string

Reply:
Maybe we should even take in postgraphile-plugin-connection-filter as a dependency and dynamically include all operators in the filtermappings...

To be able to do that, we could propose a change to https://github.com/graphile-contrib/postgraphile-plugin-connection-filter/blob/master/src/PgConnectionArgFilterOperatorsPlugin.js so that it exports the `connectionFilterScalarOperators` that includes patternMatchingOperators which in term includes: 'includes', `includesInsensitive`, `like`, `notLikeInsensitive `, etc as keys

Reply:
@christiaanwesterbeek That change makes sense. Is the idea that the user would have to include `%` explicitly if they wanted a wildcard match.

We may also look at taking in `postgraphile-plugin-fulltext-filter` as a dependency since it was mentioned in https://github.com/hjr3/ra-postgraphile/pull/2

Reply:
I believe the like vs include issue is fixed in 08d75673ed37707f10ef37745953acf865346a43